src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ILCompiler.Reflection.ReadyToRun.csproj

```diff
@@ -16,7 +16,7 @@
 
   <ItemGroup>
     <PackageReference Include="System.Reflection.Metadata" Version="1.6.0" />
```

Comment:
Does "System.Reflection.Metadata have the same problem?

I think it would be better to fix this by binplacing the crossgen2 tool into a different directory so that it does not stomp of what comes with current netcoreapp.

Reply:
> Does "System.Reflection.Metadata have the same problem?

It is not getting binplaced because the shared framework verision is higher than the package version.

> I think it would be better to fix this by binplacing the crossgen2 tool into a different directory so that it does not stomp of what comes with current netcoreapp.

R2Rdump has a transitive dependency on the System.Runtime.CompilerServices.Unsafe and is binplacing the library. I am moving it to separate directory and also removing the package references as they are redundant (Because we will always be using the shared framework version)